agnpy/spectra/tests/test_spectra.py

```diff
@@ -760,12 +760,13 @@ def test_compare_numerical_results_with_analytical_calculation(self, p, time_and
         initial_n_e = PowerLaw(k, p, gamma_min=gamma_min, gamma_max=gamma_max, mass=m_e)
         blob = Blob(1e16 * u.cm, Distance(1e27, unit=u.cm).z, 0, 10, 1 * u.G, n_e=initial_n_e)
         synch = Synchrotron(blob)
-        evaluated_n_e = initial_n_e.evaluate_time(time, synch.electron_energy_loss_per_time, subintervals_count=steps)
+        blob.n_e_time_evolution(synch.electron_energy_loss_rate, time, subintervals_count=steps)
+        evaluated_n_e = blob.n_e
```

Comment:
be careful with this, because I suspect that this will make a shallow copy of the n_e

Reply:
As we already discussed, in this case it's safe because i replace the n_e in blob at each calculation step with a new copy. But just in case the internal implementation changes in the future, i added deep copy here